network/p2p/acp118/aggregator_test.go

```diff
@@ -0,0 +1,363 @@
+				nodeID1: NewHandler(
+					&testVerifier{Errs: []*common.AppError{common.ErrUndefined}},
+					signer1,
+				),
+				nodeID2: NewHandler(
+					&testVerifier{Errs: []*common.AppError{common.ErrUndefined}},
+					signer2,
+				),
+			},
+			ctx: context.Background(),
+			validators: []Validator{
+				{
+					NodeID:    nodeID0,
+					PublicKey: pk0,
+					Weight:    1,
+				},
+				{
+					NodeID:    nodeID1,
+					PublicKey: pk1,
+					Weight:    1,
+				},
+				{
+					NodeID:    nodeID2,
+					PublicKey: pk2,
+					Weight:    1,
+				},
+			},
+			quorumNum:   2,
+			quorumDen:   3,
+			wantSigners: []int{0},
+			wantErr:     ErrFailedAggregation,
+		},
+		{
+			name: "fails aggregation from some validators - 2/3",
+			peers: map[ids.NodeID]p2p.Handler{
+				nodeID0: NewHandler(&testVerifier{}, signer0),
+				nodeID1: NewHandler(&testVerifier{}, signer1),
+				nodeID2: NewHandler(
+					&testVerifier{Errs: []*common.AppError{common.ErrUndefined}},
+					signer2,
+				),
+			},
+			ctx: context.Background(),
+			validators: []Validator{
+				{
+					NodeID:    nodeID0,
+					PublicKey: pk0,
+					Weight:    1,
+				},
+				{
+					NodeID:    nodeID1,
+					PublicKey: pk1,
+					Weight:    1,
+				},
+				{
+					NodeID:    nodeID2,
+					PublicKey: pk2,
+					Weight:    1,
+				},
+			},
+			quorumNum:   1,
+			quorumDen:   1,
+			wantSigners: []int{0, 1},
+			wantErr:     ErrFailedAggregation,
+		},
+		{
+			name: "fails aggregation from all validators",
+			peers: map[ids.NodeID]p2p.Handler{
+				nodeID0: NewHandler(
+					&testVerifier{Errs: []*common.AppError{common.ErrUndefined}},
+					signer0,
+				),
+			},
+			ctx: context.Background(),
+			validators: []Validator{
+				{
+					NodeID:    nodeID0,
+					PublicKey: pk0,
+					Weight:    1,
+				},
+			},
+			wantErr:   ErrFailedAggregation,
+			quorumNum: 1,
+			quorumDen: 1,
+		},
+		{
+			name: "context canceled",
+			peers: map[ids.NodeID]p2p.Handler{
+				nodeID0: NewHandler(&testVerifier{}, signer0),
+			},
+			ctx: func() context.Context {
+				ctx, cancel := context.WithCancel(context.Background())
+				cancel()
+
+				return ctx
+			}(),
+			validators: []Validator{
+				{
+					NodeID:    nodeID0,
+					PublicKey: pk0,
+					Weight:    1,
+				},
+			},
+			quorumNum: 0,
+			quorumDen: 1,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			client := p2ptest.NewClientWithPeers(
+				t,
+				context.Background(),
+				ids.EmptyNodeID,
+				tt.peers,
+			)
+			aggregator := NewSignatureAggregator(logging.NoLog{}, client)
+
+			unsignedMsg, err := warp.NewUnsignedMessage(
+				networkID,
+				chainID,
+				[]byte("payload"),
+			)
+			require.NoError(err)
+			msg, err := warp.NewMessage(
+				unsignedMsg,
+				&warp.BitSetSignature{Signature: [bls.SignatureLen]byte{}},
+			)
+			require.NoError(err)
+			gotMsg, gotNum, gotDen, err := aggregator.AggregateSignatures(
```

Comment:
nit: should these be `aggregatedSignatureWeight` and `totalWeight` rather than num and den which suggests numerator/denominator?

Reply:
Yeah these names make a lot more sense